src/Rules/VatValid.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Rapidez\VatValidation\Rules;
+
+use Closure;
+use Ibericode\Vat\Validator;
+use Ibericode\Vat\Vies\ViesException;
+use Illuminate\Contracts\Validation\ValidationRule;
+use Illuminate\Support\Facades\Cache;
+
+class VatValid implements ValidationRule
+{
+    /**
+     * Run the validation rule.
+     */
+    public function validate(string $attribute, mixed $value, Closure $fail): void
+    {
+        try {
+            $validator = new Validator;
+            $result = Cache::remember('vat_' . $value, config('rapidez.vatvalidation.cache_duration'), function () use ($value, $validator) {
+                return $validator->validateVatNumber($value);
+            });
+
+            if (!$result) {
+                $fail('Vat validation failed.');
+            }
+        } catch (ViesException $exception) {
+            $fail($exception->getMessage());
```

Comment:
Do we want to catch this and fail the validation, or let the error propagate and throw an error/break the endpoint?

Reply:
An error here usually represents the VAT service itself failing, so I guess it does make sense to just let it go 🤔